libraries/redcore/api/hal/helper.php

```diff
@@ -958,19 +959,20 @@ public static function getPrimaryKeysFromFields($xmlElement)
 			{
 				$fieldAttributes = self::getXMLElementAttributes($field);
 
-				if (self::isAttributeTrue($field, 'isPrimaryField'))
+				if ($primaryKeys && self::isAttributeTrue($field, 'isPrimaryField')
```

Comment:
you are missing the brackets, right now you have true || false, so it will always be true